File: python/number-of-1-bits.py
import os

class Solution:
    # @param n, an integer
    # @return an integer
    def hammingWeight(self, n):
        self.sum = 0
        n_str = '{0:032b}'.format(n)
        for char in n_str:
            self.sum = self.sum + int(char)
        return self.sum

if __name__ == '__main__':
	Solver = Solution()
	n = 11
	print "%d hamming weight is: %d" %(n, Solver.hammingWeight(n))
	os.system("pause")

        

File: c++/number-of-1-bits.cpp
#include <bitset>
#include <iostream>
#include <cstdint>
using namespace std;

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
		int tmp = n;
        std::bitset<sizeof(n)*8> x(tmp);
        for (int i=0;i<sizeof(x)*8;i++)
        {
            if (x[i]==1)
            ++count;
        }
        return count;
    }
};

int main()
{
	uint32_t n = 11;
	Solution Solver;
	int count = Solver.hammingWeight(n);
	std::cout<<n<<" hamming weight is : "<<count<<std::endl;
	system("pause");
}